### DISCUSSION TITLE
Consider Power Usage Effectiveness (PUE)

### DISCUSSION THREAD

**Discussion by Green-Software** [Green-Software-Foundation/sci#197] [SCI feedback]
The Power Usage Effectiveness (PUE) is not included as part of the operational energy. The total operational energy includes the energy consumed by the computing equipment and the supporting energy consumed by facilities (e.g. HVAC).

PUE is the ratio of total amount of energy used by a computer data center facility to the energy used by the computing equipment. If we do not include PUE, a software running at the same hardware with the same energy consumption at two different datacenters located in the same city would have the same operational energy E based on the current equation. However, if datacenter A's PUE is 1.1 while datacenter B's PUE is 1.3, the actual total energy consumption of the same software running at these two data centers would be 18% different. That difference is not reflected by the current equation.

If we consider PUE, the equation would be: SCI = ((E * PUE * I) + M) per R 

Note that there are three potential issues if we do consider PUE: (1) PUE varies by seasons. Should we use the lowest PUE or average yearly measured PUEs? (2) Large data centers (e.g. Microsoft, Google, and Facebook) monitor PUEs frequently but PUE may not be widely available in other small scale systems. Including PUE in SCI will motivate more data centers to measure and improve their PUEs. (3) If we include PUE in the equation, SCI score could be lowered without optimizing code but by optimizing facility operation or moving software to a facility with lower PUE. In general, it is still a good thing to encourage, just like we encourage software to run in a location with low carbon intensity.

**Comment by Henry-WattTime** [comment]
Oh I like this a lot!

**Comment by jawache** [comment]
That's a good point, we should make a reference to it in the energy section to include the PUE in your calculation when running in the cloud, but since the equation has to be applicable to non-cloud scenarios including it in the main equation would be confusing. I would suggest just discussing it in the `E` section. 

As far as I am aware I don't believe any cloud provider gives the PUE at the DC level, I believe all the major players either give a global yearly average or don't discuss it at all? I could be wrong about that, I know it's true for Microsoft.

**Comment by navveenb** [comment]
Including PUE is a good idea for cloud, as the data is available. For instance, Google cloud provides this quarterly, apart from yearly- https://www.google.com/about/datacenters/efficiency/. So based on the availability of PUE values, this can be reported and used.

**Comment by atg-abhishek** [comment]
One of the things that @jawache and I were talking about was that PUE needs to be strongly tied with a component boundary otherwise there is a huge potential for misuse by leaning in on specific elements that are included and not included that can skew the actual number that comes out from this. This is a great call then for standardizing what is included in the component boundary when calculating the PUE

**Comment by srini1978** [comment]
@Green-Software  I love your point about "If we include PUE in the equation, SCI score could be lowered without optimizing code but by optimizing facility operation or moving software to a facility with lower PUE. In general, it is still a good thing to encourage, just like we encourage software to run in a location with low carbon intensity" .

So if we want to calculate and compare SCI values of 2 pieces of software would including PUE in the equation skew the comparison logic ?

**Reply by jawache** [reply]
I can think of some arguments for and some against it, the question that worries me is if we allow in PUE then does that open the door for other things? Like the embodied carbon from building the DC in the first place, we previously discussed not including that but if we are rewarding DC's that have better PUEs, do we need to reward DCs that are built more sustainability in the first place?

**Reply by atg-abhishek** [reply]
And also wondering given what you mention here @jawache - does this also open the door then to market-based measures that are purchased against the consumption of the DCs?

**Comment by atg-abhishek** [comment]
1. PUE SHOULD be included under `E` 
2. Since there are no standards for its calculation and reporting (that we know of) we can provide a recommended list of things to include and if anything is excluded from that list, they MUST report why it was excluded 
3. We need to investigate also if PUE is included in some of the existing energy metrics

**Reply by jawache** [reply]
The general case here could be made that the calculation of E SHOULD represent the energy consumed from the grid to service the software rather than the energy consumed by the software running on the device.

So this means for PUE and DC use cases you would need to measure not just your energy consumption but then try to map that to the energy actually consumed from the grid.

For a non-DC use case, this also might make sense, e.g. a laptop plugged in vs. a laptop powered by a battery, even though in both use cases the energy consumption on the laptop might be the same that actual energy consumed from the grid would be different, (you lose about 10%-20% energy in charging a laptop battery, at least that's the roundtrip loss for a utility-scale battery, I'm not sure how much you lose charging a laptop/mobile battery). Conceptually it's the same as PUE in a DC.

NOTE: Google does a great job of reporting PUE by DC region, Microsoft has one number that they talk about publically and it's the global average, I'm not sure if AWS release anything publically. If we are to include this we should consider making default values available via the SCI Data project cc @dtoakley there is research out there I believe from the Uptime institute?

**Reply by jawache** [reply]
Actually maybe as a first step, this could be a disclosure requirement in the reporting rather than anything else. Feel free to calculate `E` however you want, but when you disclose your SCI number you have to acknowledge if you are taking into account PUE or not, does your `E` represent the actual energy rather than the consumed energy.

**Reply by davidmytton** [reply]
There is a standard for calculation of PUE - [ISO/IEC 30134-2:2016](https://www.iso.org/standard/63451.html) - but it's true there's no common approach to disclosure, as @jawache notes. Google is definitely the best at PUE transparency but they don't report the numbers for all their facilities, only the ones they own.

Uptime Institute has industry survey data on PUE:

https://journal.uptimeinstitute.com/is-pue-actually-going-up/
https://journal.uptimeinstitute.com/data-center-pues-flat-since-2013/

For 2020 the average is 1.59 and for 2021 it's 1.57 ([source](https://uptimeinstitute.com/2021-data-center-industry-survey-results))
